commit 9f95b17b77cf68afb6a02b1eb269957fc6b7b57b (refs/changes/40/2840/2)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-10-26T16:25:35+02:00 (1 year, 11 months ago)
    
    TOOLS-1902 'sdcadm experimental up -s UUID' should validate provided UUIDs better
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/procedures/index.js b/lib/procedures/index.js
index 94780c0..893962a 100644
--- a/lib/procedures/index.js
+++ b/lib/procedures/index.js
@@ -692,6 +692,14 @@ function coordinatePlan(opts, cb) {
             // Make sure we get a list of uuids even if hostnames may be
             // initially present as servers option:
             if (opts.servers && opts.servers.length) {
+                var notFound = opts.servers.filter(function (s) {
+                    return (!opts.serverFromUuidOrHostname[s]);
+                });
+                if (notFound.length) {
+                    next(new errors.UpdateError(format(
+                            'unknown servers "%s"', notFound.join('", "'))));
+                    return;
+                }
                 opts.servers = opts.servers.map(function (s) {
                     return opts.serverFromUuidOrHostname[s].uuid;
                 });
